akka-http-core/src/main/java/akka/http/javadsl/model/HttpMessage.java

```diff
@@ -55,6 +58,11 @@
      */
     ResponseEntity entity();
 
+    /**
+     *  Drains entity stream of this message
+     */
+    CompletionStage<Done> ignoreDataBytesJ(Materializer materializer);
```

Comment:
> Scalac detects name clash here as both have the same signature after erasure and both are in the same inheritance scope.

I was thinking about having it only declared in the super class, and trinkle through javadsl into scaladsl (would require additional super class for HttpMessage though).

> What about mixing two approaches and implementing scala method that takes implicit materializer and returns DiscardedEntity within an implicit class?

I see, I think this makes sense. Discoverability is good, since it's the same method and we're only adding the version that is able to take an implicit mat via the conversion. We'd make the conversion class be a value class such that it does not cost any allocations in run time too.

👍  please give it a shot :)